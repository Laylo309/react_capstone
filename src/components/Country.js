import { BsEyeFill } from 'react-icons/bs';
import { LinkContainer } from 'react-router-bootstrap';

const Country = (props) => {
  const { country, filter } = props;
  let ans = (
    <div className="d-flex justify-content-center align-items-center min-height">
      <div className="spinner" />
    </div>
  );

  if (country) {
    ans = Object.values(country).filter(({ name }) => (
      name.toLowerCase().startsWith(filter.toLowerCase())
    )).map((country, index) => (
      <LinkContainer key={country.id} to={`/country/${country.id}`}>
        <div
          className={`
            col-6 d-flex flex-column justify-content-between
            ${([1, 0, 0, 1][index % 4]) ? 'intense-magenta' : 'light-magenta'}
          `}
        >
          <span className="text-white mt-1 mb-4 h5">
            <BsEyeFill />
          </span>
          <span className="text-white pointer">
            <h5 className="d-inline-block m-0 text-end fw-bold">{country.name.toUpperCase()}</h5>
            <p>
              Infections:
              {' '}
              {
                Number(country.today_confirmed).toLocaleString()
              }
            </p>
          </span>
        </div>
      </LinkContainer>
    ));
  }

  return ans;
};

export default Country;
