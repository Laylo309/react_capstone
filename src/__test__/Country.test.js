import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import Country from '../components/Country';

describe('Country test', () => {
  test('The spinner appears in the Country view', () => {
    const { container } = render(<Country />);
    const linkElement = container.querySelector('.spinner');
    expect(linkElement).toBeInTheDocument();
  });

  test('Test of Snapshot', () => {
    const tree = renderer.create(<Country />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
