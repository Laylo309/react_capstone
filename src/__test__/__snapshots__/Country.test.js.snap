// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country test Test of Snapshot 1`] = `
<div
  className="d-flex justify-content-center align-items-center min-height"
>
  <div
    className="spinner"
  />
</div>
`;
